src/os/bluestore/BlueStore.cc

```diff
@@ -4600,6 +4600,15 @@ BlueStore::BlueStore(CephContext *cct,
   _init_logger();
   cct->_conf.add_observer(this);
   set_cache_shards(1);
+  if ( cct->_conf->bluestore_codel) {
+    codel = new BlueStoreSlowFastCoDel(cct,
```

Comment:
This is a memory leak since you don't release it in ~BlueStore.  Change the type of code to std::unique_ptr to avoid needing to release in ~BlueStore.

Reply:
Also, rewrap to

```
  if ( cct->_conf->bluestore_codel) {
    codel = new BlueStoreSlowFastCoDel(
      cct,
      [this](int64_t x) mutable {
	this->throttle.reset_kv_throttle_max(x);
      },
      [this]() mutable {
	return this->throttle.get_kv_throttle_current();
      });
  }
```